security/s2a/internal/crypter/testutil/common.go

```diff
@@ -0,0 +1,55 @@
+package testutil
+
+import "encoding/hex"
+
+// Constants indicating whether the test vector is valid or not.
+const (
+	ValidResult   = "valid"
+	InvalidResult = "invalid"
+)
+
+// CryptoTestVector is a struct representing a test vector for an S2AAeadCrypter
+// instance.
+type CryptoTestVector struct {
+	Id                                          int
+	Key, Plaintext, Ciphertext, Tag, Nonce, Aad []byte
+	Result, Comment                             string
+	AllocateDst                                 bool
+}
+
+// TestVector is a struct for a WycheProof test vector.
+type TestVector struct {
+	TcId    int    `json:"tcId"`
+	Comment string `json:"comment"`
+	Key     string `json:"key"`
+	Iv      string `json:"iv"`
+	Aad     string `json:"aad"`
+	Msg     string `json:"msg"`
+	Ct      string `json:"ct"`
+	Tag     string `json:"tag"`
+	Result  string `json:"result"`
+}
+
+// TestGroup is a struct for a WycheProof test group.
+type TestGroup struct {
+	IvSize  int          `json:"ivSize"`
+	KeySize int          `json:"keySize"`
+	TagSize int          `json:"tagSize"`
+	Tests   []TestVector `json:"tests"`
+}
+
+// TestFile is a struct for a WycheProof test file.
+type TestFile struct {
+	TestGroups []TestGroup `json:"testGroups"`
+}
+
+func Dehex(s string) []byte {
+	if len(s) == 0 {
+		return make([]byte, 0)
+	}
+	b, err := hex.DecodeString(s)
+	if err != nil {
+		panic(err)
```

Comment:
I think not using a panic here might make the tests more difficult to read.
Currently, we construct our test cases like this:
```
{
			Key:         testutil.Dehex("5b9604fe14eadba931b0ccf34843dab9"),
			Plaintext:   testutil.Dehex("001d0c231287c1182784554ca3a21908"),
			Ciphertext:  testutil.Dehex("26073cc1d851beff176384dc9896d5ff"),
			Tag:         testutil.Dehex("0a3ea7a5487cb5f7d70fb6c58d038554"),
			Nonce:       testutil.Dehex("028318abc1824029138141a2"),
			Result:      testutil.ValidResult,
			AllocateDst: true,
		},
```
If we return an error on Dehex, we would need to make this much longer by declaring and error checking every use of Dehex. I think panics should be avoided in real production code but I think they may be useful sometimes, particular in cases like this where the function is only used for testing.

The dehex method in ALTS also panics, and is also only used for unit tests.

Reply:
You're right. It's totally fine to use `panic` here. Since using `panic` is not very common, I wanted to take the opportunity to mention (as FYI) the Golang style guide rule of not using `panic` if possible :)